aws-android-sdk-core/src/main/java/com/amazonaws/logging/LogFactory.java

```diff
@@ -74,4 +76,24 @@ private static boolean checkApacheCommonsLoggingExists() {
             return false;
         }
     }
+
+    /**
+     * Truncate log tag to be within 23 characters in length as required by Android on certain API levels.
+     *
+     * @param logTag Log tag to be truncated
+     * @return truncated log tag
+     */
+    private static String getTruncatedLogTag(String logTag) {
+        if (logTag.length() > 23) {
+            if (checkApacheCommonsLoggingExists()) {
+                Log log = new ApacheCommonsLogging(TAG);
```

Comment:
Would this break for untruncated tags (like if our LogFactory class was subclassed or renamed to be greater than 23?

Reply:
It should not until someone renames this class or manually modifies the `TAG` property in this class. It should not break for subclasses of LogFactory with longer log tags since the `TAG` property of LogFactory is immutable.